api/v1alpha1/gateway_types.go

```diff
@@ -335,18 +335,22 @@ const (
 // GatewayCondition is an error status for a given route.
 type GatewayCondition struct {
 	// Type indicates the type of condition.
+	// +required
 	Type GatewayConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=GatewayConditionType"`
 	// Status describes the current state of this condition. Can be "True",
 	// "False", or "Unknown".
+	// +required
 	Status core.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
 	// Message is a human-understandable message describing the condition.
-	// +optional
+	// This field may be empty.
+	// +required
```

Comment:
I know this is how https://github.com/kubernetes/enhancements/pull/1624 recommends defining this field, but it sure feels less than ideal. I've left a [comment](https://github.com/kubernetes/enhancements/pull/1624) on that PR to try to get some clarification on the rationale.

Reply:
sounds good. my understanding is there is no use case to differentiate an empty string from no input from the user in reality.

I'll follow up on that thread too.